contrib/copy-tex/copy-tex.js

```diff
@@ -1,24 +1,41 @@
 import katexReplaceWithTex from './katex2tex';
 
-// Global copy handler to modify behavior on .katex elements.
-document.addEventListener('copy', function(event) {
+document.addEventListener('copy', function (event) {
     const selection = window.getSelection();
     if (selection.isCollapsed) {
-        return;  // default action OK if selection is empty
+        return; // default action OK if selection is empty
     }
-    const fragment = selection.getRangeAt(0).cloneContents();
-    if (!fragment.querySelector('.katex-mathml')) {
-        return;  // default action OK if no .katex-mathml elements
+    const range = selection.getRangeAt(0);
+
+    const sel_parent = range.commonAncestorContainer;
+    if (!(sel_parent instanceof Element)) {
+        return;
     }
-    // Preserve usual HTML copy/paste behavior.
-    const html = [];
-    for (let i = 0; i < fragment.childNodes.length; i++) {
-        html.push(fragment.childNodes[i].outerHTML);
+    const closest_katex = sel_parent.closest('.katex');
+
+    let fragment, html_contents;
+
+    if (closest_katex) {
+        // Then we are fully inside an equation. We expand the selection to be the entire equation.
+
+        fragment = /** @type Element */ (closest_katex.cloneNode(true));
+
+        if (!fragment.querySelector('.katex-mathml')) {
+            return; // default action OK if no .katex-mathml elements
+        }
+        html_contents = fragment.innerHTML;
```

Comment:
Should this be `html_contents = fragment.outerHTML`? Otherwise you're missing the `<span class="katex">` wrapper, right?

Reply:
True, thanks!

Reply:
```suggestion
        html_contents = fragment.outerHTML;
```